### ISSUE TITLE
Rename Managed DLLs to have a cor3 moniker (same as native DLLs)

### ISSUE THREAD

**Issue by rladuca** [dotnet/wpf#911]
In order to more effectively collect crash reports and differentiate them from .NET Framework reports, we want to ensure that all binaries are named differently than they are in .NET Framework.  This needs to be done by Preview 7 as this is, essentially, a binary composition and API change.

This was already done for the native assemblies (e.g. penimc_**cor3**), we should use the same suffix (**cor3**).

There are several steps to follow here:
- [ ]  Rename outputs in all projects (open and internal)
- [ ]  Update ProjectReferences in all projects (open and internal)
- [ ]  Update targets related to replacing ProjectReferences according to open/internal split
- [ ]  Update any System.Xaml/PBT code that might rely on specific names
  - [ ]  Ensure PBT works in our tests (manual and otherwise)
- [ ]  Create type forwarders for all assemblies using GenAPI (use different project names to libs)
  * These will be static and represent a snapshot in time of the API, specifically compatible with .NET 4.8
- [ ]  Update ApiCompat to ensure these forwarders are always compatible with .NET 4.8
- [ ]  Inform upstream consumers of the changes to the binaries (see [here ](https://github.com/dotnet/core-setup/issues/6694)for an example).

This list will update as things progress.

**Comment by rladuca** [comment]
Working branch: https://github.com/dotnet/wpf/tree/dev/roladuca/assemblyrename

**Comment by rladuca** [comment]
After discussions on the consequences of doing this in .NET Core 3, we've decided to move this off to .NET 5.  The reasoning is that we'd like to make a clean break not just in naming, but also in form of the assemblies.

That change is too large to do at this moment and we don't yet know the exact shape of it that we want.  I am going to move this off to NET 5 as a marker for that work so when we sweep .NET 5 tasks we can revive the discussions.

So, for now, we've decided that we're not going to worry about crash dumps being as easily differentiated.  There are, potentially, other avenues to tightening those queries we can explore for the time being.

**Comment by mikedn** [comment]
> So, for now, we've decided that we're not going to worry about crash dumps being as easily differentiated. 

Perhaps you can just rename some internal class/method that is likely to be always in the callstack - e.g. something Dispatcher related. Then if you look at the callstack you can tell if it's Desktop or Core.

In any case, adding suffixes like `_cor3` to managed assemblies is rather unusual.

**Comment by rladuca** [comment]
@mikedn Yes, we decided that, while it's been the way we've done things with native assemblies, we shouldn't apply this to the managed assemblies.

We're taking a look at some various ways we can figure out WER reports so that they're differentiated from .NET Framework, which is the main driver for doing something like this.  We'll have to come up with some way of querying the differences, I'm not sure a deeper item on the stack would work for us since it depends on what we can pivot on easily in queries.  In any case, we'll be looking into it as we move toward release.

**Comment by rladuca** [comment]
Just to add, the native assemblies are already renamed from _v0300 and _v0400, so we can easily determine what a dump is targeting in the manual sense.  It's just about digging into crash reports and getting good automated bucketing there.
